### ISSUE TITLE
Install 2.4.6-p8 (also 2.4.7-p3) Unable to apply patch

### ISSUE THREAD

**Issue by hdezela** [magento/magento2#39514]
### Preconditions and environment

- Magento 2.4.6-p8 / 2.4.7-p3
- ArchLinux December 2024
- PHP 8.2.26
- Nginx 1.26.2
- MariaDB 10.6.20
- Redis 7.2.5
- Opensearch 2.12.0
- Composer 2.7.9

### Steps to reproduce

1. composer create-project --repository-url=https://repo.magento.com/ magento/project-community-edition=2.4.6-p8
2. bin/magento setup:install --base-url=https://[redacted] --db-host=localhost --db-name=[redacted] --db-user=[redacted] --db-password=[redacted] --admin-firstname=[redacted] --admin-lastname=[redacted] --admin-email=[redacted] --admin-user=[redacted] --admin-password=[redacted] --language=es_ES --currency=CLP --timezone=America/Santiago --use-rewrites=1 --search-engine=opensearch --opensearch-host=[redacted] --opensearch-port=[redacted] --opensearch-index-prefix=magento2 --opensearch-timeout=15

### Expected result

Magento installs

### Actual result

Installation fails at:

[Progress: 115 / 1331]
Module 'Magento_InventorySales':

with error:

In PatchApplier.php line 251:

Unable to apply patch Magento\InventorySales\Setup\Patch\Schema\InitializeWebsiteDefaultSock for module Magento_InventorySales. Original exception message: The default website isn't defined. Set the website and try again.

![Screenshot 2024-12-30 011935](https://github.com/user-attachments/assets/b3868de6-3725-49fd-aa60-7fe24f189126)

### Additional information

I have tried all the fixes shown in [this issue](https://github.com/magento/magento2/issues/20034) but none of them work. I've also tried the patches supplied in [this thread](https://magento.stackexchange.com/questions/315644/unable-to-apply-patch-magento-inventorysales-setup-patch-schema-initializewebsite) and it also does not work.

I've tried disabling the module but it locks out due to dependencies, and patching it to omit the step creates another chain of errors. Also hardcoding the default website in the code so it's not looked up also fails.

### Triage and priority

- [X] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by engcom-Bravo** [comment]
Hi @hdezela,

Thanks for your reporting and collaboration.

We have tried to install Magento 2.4.6-p8 and 2.4.7-p3 and we are able to install successfully.Kindly refer the screenshots.


<img width="1416" alt="Screenshot 2025-01-08 at 10 27 34 am" src="https://github.com/user-attachments/assets/2d2471b9-b7c3-49e0-a737-43f6adf08966" />


Kindly check your system requirements https://experienceleague.adobe.com/en/docs/commerce-operations/installation-guide/system-requirements and let us know if you are still facing any issue.

Thanks.

**Comment by hdezela** [comment]
I have tried again, reconfirming all requirements and made a [step by step video](https://youtu.be/0UFPvz_iaR0) showing everything and it still gives me the exact same error at the same step. I'm attaching my configs for the infra and also a dump of the db as it sits (right after the error) to maybe help troubleshoot this.

I don't know what else I can say or do, I clearly see your evidence that you can install it but I also clearly see my evidence that I cannot.

I did try installing an earlier version (2.4.5) and upgrading that to the latest 2.4.7-p3 and it is working fine, so I can at least confirm that the setup supports Magento running on it, just not a clean install.

[configs.tgz](https://github.com/user-attachments/files/18351030/configs.tgz)
[magento.sql.gz](https://github.com/user-attachments/files/18351031/magento.sql.gz)

**Comment by hdezela** [comment]
After diving around and echoing everything from everywhere I was able to find a common error that pops up when using PHP 8.2 or higher:

`[2025-01-11T18:09:09.392879+00:00] main.ERROR: There is already an active transaction [] []
[2025-01-11T18:09:09.393407+00:00] main.CRITICAL: PDOException: There is already an active transaction in /home/magento/code/vendor/magento/zendframework1/library/Zend/Db/Adapter/Pdo/Abstract.php:305
Stack trace:
#0 /home/magento/code/vendor/magento/zendframework1/library/Zend/Db/Adapter/Pdo/Abstract.php(305): PDO->beginTransaction()
#1 /home/magento/code/vendor/magento/zendframework1/library/Zend/Db/Adapter/Abstract.php(496): Zend_Db_Adapter_Pdo_Abstract->_beginTransaction()
#2 /home/magento/code/vendor/magento/framework/DB/Adapter/Pdo/Mysql.php(297): Zend_Db_Adapter_Abstract->beginTransaction()
#3 /home/magento/code/vendor/magento/framework/Model/ResourceModel/AbstractResource.php(66): Magento\Framework\DB\Adapter\Pdo\Mysql->beginTransaction()
#4 /home/magento/code/vendor/magento/framework/Model/ResourceModel/Db/AbstractDb.php(380): Magento\Framework\Model\ResourceModel\AbstractResource->beginTransaction()
#5 /home/magento/code/vendor/magento/framework/Model/AbstractModel.php(658): Magento\Framework\Model\ResourceModel\Db\AbstractDb->save()
#6 /home/magento/code/vendor/magento/module-theme/Model/Theme/Plugin/Registration.php(116): Magento\Framework\Model\AbstractModel->save()
#7 /home/magento/code/vendor/magento/module-theme/Model/Theme/Plugin/Registration.php(84): Magento\Theme\Model\Theme\Plugin\Registration->updateThemeData()
#8 /home/magento/code/vendor/magento/framework/Interception/Interceptor.php(121): Magento\Theme\Model\Theme\Plugin\Registration->beforeExecute()
#9 /home/magento/code/vendor/magento/framework/Interception/Interceptor.php(153): Magento\Backend\Controller\Adminhtml\Auth\Login\Interceptor->Magento\Framework\Interception\{closure}()
#10 /home/magento/code/generated/code/Magento/Backend/Controller/Adminhtml/Auth/Login/Interceptor.php(23): Magento\Backend\Controller\Adminhtml\Auth\Login\Interceptor->___callPlugins()
#11 /home/magento/code/vendor/magento/framework/App/Action/Action.php(111): Magento\Backend\Controller\Adminhtml\Auth\Login\Interceptor->execute()
#12 /home/magento/code/vendor/magento/module-backend/App/AbstractAction.php(151): Magento\Framework\App\Action\Action->dispatch()
#13 /home/magento/code/vendor/magento/framework/Interception/Interceptor.php(58): Magento\Backend\App\AbstractAction->dispatch()
#14 /home/magento/code/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Backend\Controller\Adminhtml\Auth\Login\Interceptor->___callParent()
#15 /home/magento/code/vendor/magento/module-backend/App/Action/Plugin/Authentication.php(145): Magento\Backend\Controller\Adminhtml\Auth\Login\Interceptor->Magento\Framework\Interception\{closure}()
#16 /home/magento/code/vendor/magento/framework/Interception/Interceptor.php(135): Magento\Backend\App\Action\Plugin\Authentication->aroundDispatch()
#17 /home/magento/code/vendor/magento/framework/Interception/Interceptor.php(153): Magento\Backend\Controller\Adminhtml\Auth\Login\Interceptor->Magento\Framework\Interception\{closure}()
#18 /home/magento/code/generated/code/Magento/Backend/Controller/Adminhtml/Auth/Login/Interceptor.php(32): Magento\Backend\Controller\Adminhtml\Auth\Login\Interceptor->___callPlugins()
#19 /home/magento/code/vendor/magento/framework/App/FrontController.php(245): Magento\Backend\Controller\Adminhtml\Auth\Login\Interceptor->dispatch()
#20 /home/magento/code/vendor/magento/framework/App/FrontController.php(212): Magento\Framework\App\FrontController->getActionResponse()
#21 /home/magento/code/vendor/magento/framework/App/FrontController.php(146): Magento\Framework\App\FrontController->processRequest()
#22 /home/magento/code/vendor/magento/framework/Interception/Interceptor.php(58): Magento\Framework\App\FrontController->dispatch()
#23 /home/magento/code/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\App\FrontController\Interceptor->___callParent()
#24 /home/magento/code/vendor/magento/framework/Interception/Interceptor.php(153): Magento\Framework\App\FrontController\Interceptor->Magento\Framework\Interception\{closure}()
#25 /home/magento/code/generated/code/Magento/Framework/App/FrontController/Interceptor.php(23): Magento\Framework\App\FrontController\Interceptor->___callPlugins()
#26 /home/magento/code/vendor/magento/framework/App/Http.php(116): Magento\Framework\App\FrontController\Interceptor->dispatch()
#27 /home/magento/code/vendor/magento/framework/App/Bootstrap.php(264): Magento\Framework\App\Http->launch()
#28 /home/magento/code/pub/index.php(30): Magento\Framework\App\Bootstrap->run()
#29 {main} {"report_id":"b7a3d2710a41fba8d43a0ae4aeef9851f089c88a21ef7988ab98eb5fea2728ba"} []`

Apparently there has been a change from PHP 8.2 and above in the PDO controller in that it treats commits in some special way (beyond my understanding). This causes Magento to think that a transaction is still pending (because that's what PDO is telling it) but the transaction has actually been executed.

Changing the mysql setting "_autocommit = 0_" to "_autocommit = 1_" fixes the Magento 2.4.6+ installation issue and some other weird issues I've seen with those versions.

If you have a "tuned" mysql config from earlier versions that recommended disabling autocommit you will come upon this error.

So basically, if you're upgrading or installing to a Magento 2.4.6 or higher, you need to double-check if autocommit is enabled in your mysql config.
